static/js/hoc/withNavSidebar.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import React from "react"
+
+import Sidebar from "../components/Sidebar"
+import Navigation from "../components/Navigation"
+
+const withNavSidebar = (WrappedComponent: Class<React.Component<*, *, *>>) => {
+  class WithNavSidebar extends React.Component {
+    render() {
+      const { subscribedChannels, location: { pathname } } = this.props
+
+      return (
+        <div className="content">
+          <Sidebar>
+            <Navigation subscribedChannels={subscribedChannels} pathname={pathname} />
+          </Sidebar>
+          <div className="main-content">
+            <WrappedComponent {...this.props} />
+          </div>
+        </div>
+      )
+    }
+  }
+  return WithNavSidebar
+}
+
+export default withNavSidebar
```

Comment:
Yeah, it is a hacky approach overall, we discussed various ways to set this up but given the limitations of the material drawer this is what we decided on. And yeah, the navigation component will be rendered twice, I'm not particularly worried about that duplication, but if you're concerned we can try to add a way to avoid it.

Reply:
Unless there's a really compelling reason not to, I think we should just go ahead and avoid rendering `Sidebar` & `Navigation` when the `showDrawer` prop is true. The pattern of creating a variable (e.g.: `renderedThing`) and setting its value as some rendered component under certain conditions is something we do in a few places in MM. I don't see why we can't do it here. I see 2 main drawbacks leaving it as-is: the extra rendering, and the potential confusion around seeing this component rendered in code and not seeing it on the page under certain conditions.

Reply:
I don't think that we can use the `showDrawer` prop to decide whether we need to render the `<Sidebar />` component - `showDrawer` is only true when the user has clicked the 'hamburger' button.

The double render is happening in the first place because of a limitation of the `<Drawer />` component, I had wanted originally to have a div on the page which was a static navigation component on desktop and acted as a drawer on mobile, but because of the DOM structure that we need to use with the `<Drawer />` that didn't quite work out. So, because of that, the `<Navigation />` component has to be rendered at two different points in the component hierarchy. We want to show one of those places on mobile (the `<Drawer />`) and one on desktop (the `<Sidebar />`). There's two ways, that I see, to accomplish that:

- render both and hide one or the other with media queries
- use JS to detect the viewport width and then decide which to render

I prefer the first solution (the current implementation) because I'm just a little sketched out about trying to detect screen width - we tried to use a React package that did this in MM and ran into some pretty serious issues with testing and so on. I think also to me rendering one HTML structure and then using a built-in browser feature (media queries) to hide what we need feels like a more standard 'responsive' approach to me, if that makes sense.

Does that make sense overall? IMO the current approach is the best workaround for the limitation of the `<Drawer />` component as it is implemented, I agree it's not optimal. I would have preferred to implement something from scratch that did exactly what we needed but this is what we're doing so `¯\_(ツ)_/¯`

Reply:
Makes sense. That was a misunderstanding on my part. This is definitely low-impact enough that we can leave as-is and think of something more elegant (if that exists) in the future